storage: TestLogSplits failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/19860c9f7b02a698109ff3905b93c384cdd8d6d5

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLogSplits PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1090355&tab=buildLog

```
I190112 06:26:01.009672 96798 storage/stores.go:243  [n1] read 0 node addresses from persistent storage
I190112 06:26:01.009881 96798 server/node.go:697  [n1] connecting to gossip network to verify cluster ID...
I190112 06:26:01.009922 96798 server/node.go:722  [n1] node connected via gossip and verified as part of cluster "8cace3a7-8cc8-4089-b490-6ee1b1ec1f0d"
I190112 06:26:01.009942 96798 server/node.go:546  [n1] node=1: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I190112 06:26:01.010165 96798 server/status/recorder.go:610  [n1] available memory from cgroups (8.0 EiB) exceeds system memory 16 GiB, using system memory
I190112 06:26:01.010184 96798 server/server.go:1862  [n1] Could not start heap profiler worker due to: directory to store profiles could not be determined
I190112 06:26:01.010258 96798 server/server.go:1559  [n1] starting https server at 127.0.0.1:45973 (use: 127.0.0.1:45973)
I190112 06:26:01.010274 96798 server/server.go:1561  [n1] starting grpc/postgres server at 127.0.0.1:35059
I190112 06:26:01.010290 96798 server/server.go:1562  [n1] advertising CockroachDB node at 127.0.0.1:35059
I190112 06:26:01.085246 96850 sql/event_log.go:135  [n1,intExec=optInToDiagnosticsStatReporting] Event: "set_cluster_setting", target: 0, info: {SettingName:diagnostics.reporting.enabled Value:true User:root}
I190112 06:26:01.099936 97241 sql/event_log.go:135  [n1,intExec=set-setting] Event: "set_cluster_setting", target: 0, info: {SettingName:version Value:2.1-4 User:root}
I190112 06:26:01.116721 97285 sql/event_log.go:135  [n1,intExec=disableNetTrace] Event: "set_cluster_setting", target: 0, info: {SettingName:trace.debug.enable Value:false User:root}
I190112 06:26:01.136149 97275 sql/event_log.go:135  [n1,intExec=initializeClusterSecret] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.secret Value:e1625a13-6ca1-4908-9252-849365a0bada User:root}
I190112 06:26:01.153028 97282 sql/event_log.go:135  [n1,intExec=create-default-db] Event: "create_database", target: 50, info: {DatabaseName:defaultdb Statement:CREATE DATABASE IF NOT EXISTS defaultdb User:root}
I190112 06:26:01.162666 97296 sql/event_log.go:135  [n1,intExec=create-default-db] Event: "create_database", target: 51, info: {DatabaseName:postgres Statement:CREATE DATABASE IF NOT EXISTS postgres User:root}
I190112 06:26:01.168686 96798 server/server.go:1620  [n1] done ensuring all necessary migrations have run
I190112 06:26:01.168709 96798 server/server.go:1623  [n1] serving sql connections
I190112 06:26:01.172717 97227 server/server_update.go:66  [n1] no need to upgrade, cluster already at the newest version
I190112 06:26:01.189414 97268 storage/replica_command.go:349  [n1,split,s1,r19/1:/{Table/23-Max}] initiating a split of this range at key /Table/24 [r20]
I190112 06:26:01.192726 97230 sql/event_log.go:135  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:127.0.0.1:35059 Attrs: Locality: ServerVersion:2.1-4 BuildTag:v2.2.0-alpha.20181217-551-g19860c9 StartedAt:1547274361005456469 LocalityAddress:[] XXX_NoUnkeyedLiteral:{} XXX_sizecache:0} ClusterID:8cace3a7-8cc8-4089-b490-6ee1b1ec1f0d StartedAt:1547274361005456469 LastUp:1547274361005456469}
I190112 06:26:01.194442 96798 storage/replica_command.go:349  [n1,s1,r5/1:/{System/tse-Table/System…}] initiating a split of this range at key "splitkey" [r21]
I190112 06:26:01.205698 96798 util/stop/stopper.go:536  quiescing; tasks left:
1      [async] storage.merge: processing replica
1      [async] closedts-rangefeed-subscriber
I190112 06:26:01.205837 96225 rpc/nodedialer/nodedialer.go:90  [n1,merge,s1,r5/1:{/System/tse-splitkey}] unable to connect to n1: initial connection heartbeat failed: rpc error: code = Canceled desc = context canceled
I190112 06:26:01.205926 96798 util/stop/stopper.go:536  quiescing; tasks left:
1      [async] closedts-rangefeed-subscriber
--- FAIL: TestLogSplits (0.25s)
	log_test.go:66: expected 1 splits, found 2

```

--- a/pkg/storage/log_test.go
+++ b/pkg/storage/log_test.go
@@ -61,9 +61,11 @@ func TestLogSplits(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	// verify that every the count has increased by one.
-	if a, e := countSplits(), initialSplits+1; a != e {
-		t.Fatalf("expected %d splits, found %d", e, a)
+	// Verify that the count has increased by at least one. Realistically it's
+	// almost always by exactly one, but if there are any other splits they
+	// might race in after the previous call to countSplits().
+	if now := countSplits(); now <= initialSplits {
+		t.Fatalf("expected >= %d splits, found %d", initialSplits, now)
 	}
 
 	// verify that RangeID always increases (a good way to see that the splits

